backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interest/InterestComputationCommand.java

```diff
@@ -186,24 +185,26 @@ private InitialInterestAllocationResult createInterestRecords(
 				}
 			}
 
-			saveAddedValueInterestRecords(currentInvoiceAllocations);
-			totalInterestScore = totalInterestScore.add(currentInvoiceAllocations.getAllocatedInterestScore());
-			splitOpenAmount(currentInvoiceAllocations);
+			saveAddedValueInterestRecords(currentInterimContractAllocations);
+			totalInterestScore = totalInterestScore.add(currentInterimContractAllocations.getAllocatedInterestScore());
+			//splitOpenAmount(currentInterimContractAllocations);
 		}
 
-		saveSubtractedValueInterestRecords(request,
-				settings.getBonusInterestRate(),
-				shippingNotificationIterator,
-				currentShippingNotification);
-
+		if (!interimInvoiceExists)
+		{
+			saveSubtractedValueInterestRecords(request,
+											   settings.getBonusInterestRate(),
+											   shippingNotificationIterator,
+											   currentShippingNotification);
+		}
 		return InitialInterestAllocationResult.of(totalInterestScore);
 	}
 
 	private void saveSubtractedValueInterestRecords(
 			@NonNull final InterestComputationRequest request,
 			@NonNull final Percent bonusInterestRate,
 			@NonNull final Iterator<ModularContractLogEntry> shippingNotificationIterator,
-			@Nullable final InvoiceAllocations.AllocationItem currentShippingNotification)
+			@Nullable final InterimContractAllocations.AllocationItem currentShippingNotification)
 	{
```

Comment:
Looks like we could have a new request that shall be given as param and shall contain InterestComputationRequest, bonusInterestRate and InterimContractAllocations.AllocationItem.
And saw that there is other method bellow that could benefit of this, as well.